### PR TITLE
py-py7zr: update to 0.18.5

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.3.1 21E258 x86_64
Xcode 13.3.1 13E500a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@reneeotten maybe you have any idea why it had failed on tests?

**Comment by reneeotten** [comment]
> @reneeotten maybe you have any idea why it had failed on CI?

it tells you that you're missing a dependency on `py-wheel`. In this case since upstream has a `pyproject.toml` file is to set `python.pep517 yes` which will use by default the `setuptools` backend. You can then remove the explicit build dependencies from the portfile. That should do the trick I think.

**Comment by catap** [comment]
@reneeotten I don't think so. This port is built for all 3 python version on my laptop where I haven't got anything in python38; Thus, clean virtual machine with 10.15 also allows to build it.

**Comment by catap** [comment]
Yay, I understand it.

I mad `port test XXX` and after that `port install XXX` which works, but `port install XXX` does.

You're suggestion works. Thanks.

File: python/py-py7zr/Portfile [hunk 1]
```diff
@@ -16,9 +16,11 @@ long_description    ${description}
...
 homepage            https://github.com/miurahr/py7zr
 
-checksums           rmd160  65cbfd2ce4b44ed8cae54c17ae1e64a5c02d6db7 \
-                    sha256  6986e90ca4ca1f4999d57987a54d891673c9f8befa02c9c6cd8201893a17442c \
-                    size    4255058
+checksums           rmd160  e71df6f0fd74398f9547ad28a0a1fad93fa9497b \
+                    sha256  81ebb10619739764c2d80ef222e5845373ce84523183371ae9cd88e0df908929 \
+                    size    4260349
+
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-py7zr/Portfile:23]: when adding this you can remove remove to explicit build dependency on `setuptools` below as this added automatically by the `python` PG (you will need the `setuptools_scm` dependency though)

**Comment by reneeotten** [comment]
> Yay, I understand it.
> 
> I mad `port test XXX` and after that `port install XXX` which works, but `port install XXX` does.

that's because you have "wheel" listed as a test dependency )for reasons I do not understand btw). So running the test first will i stall the package and then the build will succeed. 

However that's just a "lucky" side effect and still means that the dependencies were not specified correctly. It is always recommended to build in trace-mode (i.e., adding the "-t" flag) so that only dependencies that you have listed will be used regardless if they are installed for some other reason (like running the tests first). 

You can likely prune the test dependencies as well - you't not i interested in coverage and such so that van be disabled.

File: python/py-py7zr/Portfile (hunk 1 above)
Comment (catap) [reply python/py-py7zr/Portfile:23]: @reneeotten it's done

**Comment by catap** [comment]
@reneeotten cleaned up test depenendcies.

File: python/py-py7zr/Portfile (hunk 1 above)
Comment (reneeotten) [reply python/py-py7zr/Portfile:23]: ehm, I still see `port:py${python.version}-setuptools` listed below under `depdends_build-append`

Comment (catap) [reply python/py-py7zr/Portfile:23]: @reneeotten indeed, forgotten push. Sorry.

**Pull request merged**